test/autoproj/daemon/test_overrides_retriever.rb

```diff
@@ -116,143 +99,117 @@ def create_pull_request(owner, name, number, body, state: "open")
                     pr_driver_orogen_gps_ublox = add_pull_request(
-                        "tidewise", "drivers-orogen-gps_ublox",
-                        22, body_driver_orogen_gps_ublox
+                        "tidewise/drivers-orogen-gps_ublox", 22,
+                        body_driver_orogen_gps_ublox
+                    )
+                    pr_package_set = add_pull_request(
+                        "tidewise/tidewise.common-package_set", 44, "package_set"
                     )
 
-                    depends = retriever.retrieve_dependencies(pr_drivers_gps_ublox)
-                    assert_equal [pr_driver_orogen_gps_ublox], depends
+                    depends = retriever.dependencies(pr_driver_orogen_gps_ublox)
+                    assert_equal [pr_package_set], depends
                 end
-                it "does not add same PR twice" do
+
+                it "recognizes that pull requests are different if they have "\
+                   "different URLs" do
                     body_driver_gps_ublox = <<~EOFBODY
                         Depends on:
-                        - [ ] tidewise/drivers-orogen-gps_ublox#22
-                        - [ ] rock-core/base-cmake#44
+                        - [ ] https://gitlab.com/tidewise/drivers-orogen-gps_ublox/merge_requests/22
+                        - [ ] https://github.com/tidewise/drivers-orogen-gps_ublox#22
                     EOFBODY
                     pr_drivers_gps_ublox = add_pull_request(
-                        "tidewise", "drivers-gps_ublox",
-                        11, body_driver_gps_ublox
+                        "tidewise/drivers-gps_ublox", 22, body_driver_gps_ublox
                     )
-
-                    body_driver_orogen_gps_ublox = <<~EOFBODY
-                        Depends on:
-                        - [ ] rock-core/base-cmake#44
-                    EOFBODY
-                    pr_driver_orogen_gps_ublox = add_pull_request(
-                        "tidewise", "drivers-orogen-gps_ublox",
-                        22, body_driver_orogen_gps_ublox
+                    gitlab = add_pull_request(
+                        "tidewise/drivers-orogen-gps_ublox", 22,
+                        service: "gitlab.com"
                     )
-
-                    pr_base_cmake = add_pull_request(
-                        "rock-core", "base-cmake",
-                        44, nil
+                    github = add_pull_request(
+                        "tidewise/drivers-orogen-gps_ublox", 22
                     )
 
-                    depends = retriever.retrieve_dependencies(pr_drivers_gps_ublox)
-                    assert_equal [pr_driver_orogen_gps_ublox,
-                                  pr_base_cmake], depends
+                    depends = retriever.dependencies(pr_drivers_gps_ublox)
+                    assert_equal Set[gitlab, github], depends.to_set
                 end
-                it "allows depending on pull requests from different services" do
```

Comment:
Could you keep this test case, please?

Reply:
I kept it. It's just under #dependencies since #retrieve_dependencies does not exist anymore